timemachine/cpp/src/kernels/k_nonbonded.cuh

```diff
@@ -5,6 +5,15 @@
 #include "nonbonded_common.cuh"
 #include "surreal.cuh"
 
+// Divide [0,1]^3 box into HILBERT_GRID_DIM^3 voxels for Hilbert sort
+static const int HILBERT_GRID_DIM = 128;
+
+// Encode grid index along each dimension using HILBERT_N_BITS
+static const int HILBERT_N_BITS = 8;
```

Comment:
Good call changing to static assertions! Small nit: would it make sense for the assertions to live beside the implementation, rather than in the header file? To me this seems more appropriate, since the assertions are specific to the implementation. Admittedly it's also nice to have the assertions near the definition..

Reply:
I think I'd prefer to keep these in the header -- to be right next to the definition as you mention, and to avoid 2x duplication if these were in the implementations!